### ISSUE TITLE
search breaks when EASYBUILD_TRY_TOOLCHAIN_VERSION is set

### ISSUE THREAD

**Issue by dominikborkowski** [hpcugent/easybuild-framework#1371]
It seems that setting the environment variable EASYBUILD_TRY_TOOLCHAIN_VERSION affects negatively the search function (but it works just fine when building software).

Example:
* working:
```
dom@sfxlogin3:/apps/easybuild> eb -S ngs
== temporary log file in case of crash /tmp/eb-tJSfPg/easybuild-TPk5ci.log
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo-vbi/shadowfax-sandy_bridge
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo-vbi/generic
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo/shadowfax-sandy_bridge
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo-vbi/shadowfax-sandy_bridge
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo-vbi/generic
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo/shadowfax-sandy_bridge
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/software/shadowfax-sandy_bridge/EasyBuild/2.2.0/lib64/python2.6/site-packages/easybuild_easyconfigs-2.2.0-py2.6.egg/easybuild/easyconfigs
CFGS1=/apps/easybuild/software/shadowfax-sandy_bridge/EasyBuild/2.2.0/lib64/python2.6/site-packages/easybuild_easyconfigs-2.2.0-py2.6.egg/easybuild/easyconfigs/n/ngspice
 * $CFGS1/ngspice-26-intel-2014b.eb
== temporary log file(s) /tmp/eb-tJSfPg/easybuild-TPk5ci.log* have been removed.
== temporary directory /tmp/eb-tJSfPg has been removed.
```
* broken:
```
dom@sfxlogin3:/apps/easybuild> export EASYBUILD_TRY_TOOLCHAIN_VERSION=1.7.20
dom@sfxlogin3:/apps/easybuild> eb -S ngs
== temporary log file in case of crash /tmp/eb-N_tMcj/easybuild-N5D92K.log
== Searching (case-insensitive) for 'ngs' in /tmp/eb-N_tMcj/tweaked_easyconfigs
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo-vbi/shadowfax-sandy_bridge
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo-vbi/generic
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo/shadowfax-sandy_bridge
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo-vbi/shadowfax-sandy_bridge
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo-vbi/generic
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/ebfiles_repo/shadowfax-sandy_bridge
== Searching (case-insensitive) for 'ngs' in /apps/easybuild/software/shadowfax-sandy_bridge/EasyBuild/2.2.0/lib64/python2.6/site-packages/easybuild_easyconfigs-2.2.0-py2.6.egg/easybuild/easyconfigs
CFGS1=/apps/easybuild/software/shadowfax-sandy_bridge/EasyBuild/2.2.0/lib64/python2.6/site-packages/easybuild_easyconfigs-2.2.0-py2.6.egg/easybuild/easyconfigs/n/ngspice
 * $CFGS1/ngspice-26-intel-2014b.eb
Traceback (most recent call last):
  File "/apps/easybuild/software/shadowfax-sandy_bridge/EasyBuild/2.2.0/lib64/python2.6/site-packages/easybuild_framework-2.2.0-py2.6.egg/easybuild/main.py", line 355, in <module>
    main()
  File "/apps/easybuild/software/shadowfax-sandy_bridge/EasyBuild/2.2.0/lib64/python2.6/site-packages/easybuild_framework-2.2.0-py2.6.egg/easybuild/main.py", line 276, in main
    easyconfigs = tweak(easyconfigs, build_specs, targetdir=tweaked_ecs_path)
  File "/apps/easybuild/software/shadowfax-sandy_bridge/EasyBuild/2.2.0/lib64/python2.6/site-packages/easybuild_framework-2.2.0-py2.6.egg/easybuild/framework/easyconfig/tweak.py", line 102, in tweak
    toolchain = orig_ecs[-1]['ec']['toolchain']
IndexError: list index out of range
```

**Comment by boegel** [comment]
@dominikborkowski: that's definitely a bug, thanks for reporting this, I'll see how this can be fixed.

**Comment by riccardomurri** [comment]
Same problem here:

        $ eb --search libtool --try-toolchain=foss,2015a
        == temporary log file in case of crash /tmp/eb-Zyp2HA/easybuild-J_GOBn.log
        == Searching (case-insensitive) for 'libtool' in /tmp/eb-Zyp2HA/tweaked_easyconfigs
        [...]
         * /apps/redhatenterpriseserver-7.1-x86_64/software/EasyBuild/2.7.0-foss-2015a/lib/python2.7/site-packages/easybuild_easyconfigs-2.7.0-py2.7.egg/easybuild/easyconfigs/l/libtool/libtool-2.4.6-intel-2016a.eb
         * /apps/redhatenterpriseserver-7.1-x86_64/software/EasyBuild/2.7.0-foss-2015a/lib/python2.7/site-packages/easybuild_easyconfigs-2.7.0-py2.7.egg/easybuild/easyconfigs/l/libtool/libtool-2.4.6.eb
        Traceback (most recent call last):
          File "/usr/lib64/python2.7/runpy.py", line 162, in _run_module_as_main
            "__main__", fname, loader, pkg_name)
          File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
            exec code in run_globals
          File "/net/nfs4/nfs.hydra/apps/redhatenterpriseserver-7.1-x86_64/software/EasyBuild/2.7.0-foss-2015a/lib/python2.7/site-packages/easybuild_framework-2.7.0-py2.7.egg/easybuild/main.py", line 408, in <module>
            main()
          File "/net/nfs4/nfs.hydra/apps/redhatenterpriseserver-7.1-x86_64/software/EasyBuild/2.7.0-foss-2015a/lib/python2.7/site-packages/easybuild_framework-2.7.0-py2.7.egg/easybuild/main.py", line 325, in main
            easyconfigs = tweak(easyconfigs, build_specs, targetdir=tweaked_ecs_path)
          File "/apps/redhatenterpriseserver-7.1-x86_64/software/EasyBuild/2.7.0-foss-2015a/lib/python2.7/site-packages/easybuild_framework-2.7.0-py2.7.egg/easybuild/framework/easyconfig/tweak.py", line 102, in tweak
            toolchain = orig_ecs[-1]['ec']['toolchain']
        IndexError: list index out of range

It looks like EB continues in the `main` function after having done the search.

Or this can also be seen as a bug in the `tweak` function that expects a non-empty list.

**Comment by boegel** [comment]
fixed in #1937
